crates/pow/src/lib.rs

```diff
@@ -49,22 +48,31 @@ pub struct PoWRateLimiterConfig {
 }
 
 impl PoWRateLimiter {
-    /// Creates a new `PoW` instance and starts a `tokio` task to clean up expired challenges.
+    /// Creates a new `PoW` instance and starts a `tokio` task that periodically cleans up expired
+    /// challenges.
     #[cfg(feature = "tokio")]
     pub fn new_with_cleanup(secret: [u8; 32], config: PoWRateLimiterConfig) -> Self {
-        let challenge_cache = Arc::new(RwLock::new(ChallengeCache::new(config.challenge_lifetime)));
+        let cleanup_interval = config.cleanup_interval;
+        let rate_limiter = Self::new(secret, config);
+        let challenge_cache = rate_limiter.challenges.clone();
 
-        // Start the cleanup task
-        let cleanup_state = challenge_cache.clone();
         tokio::spawn(async move {
-            Self::run_cleanup(cleanup_state, config.cleanup_interval).await;
+            let mut interval = tokio::time::interval(cleanup_interval);
+
+            loop {
+                interval.tick().await;
+                let current_time = SystemTime::now()
+                    .duration_since(UNIX_EPOCH)
+                    .expect("current timestamp should be greater than unix epoch")
+                    .as_secs();
+                challenge_cache
+                    .write()
+                    .expect("challenge cache lock should not be poisoned")
+                    .drop_expired_challenges(current_time);
```

Comment:
I would convert this into a small helper function to reduce code duplication - e.g., something like:
```rust
fn cleanup_challenge_cache(challenge_cache: Arc<RwLock<ChallengeCache>>) {
    let current_time = SystemTime::now()
        .duration_since(UNIX_EPOCH)
        .expect("current timestamp should be greater than unix epoch")
        .as_secs();
   challenge_cache
        .write()
        .expect("challenge cache lock should not be poisoned")
        .drop_expired_challenges(current_time);
}
```